### ISSUE TITLE
Hidden columns in the Containers tab table re-enable themselves

### ISSUE THREAD

**Issue by DuckThom** [docker/for-mac#6391]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 83365C73-D17A-4032-8408-BEA3888B7EBC/20220705113001

### Expected behavior

When hiding a column, the column should stay hidden

### Actual behavior

The column reappears/re-enables itself after a couple of seconds

![image](https://user-images.githubusercontent.com/6512901/177319487-8293a044-4a84-489f-ac91-df4cd9bd3a57.png)

https://user-images.githubusercontent.com/6512901/177319689-20f2a58a-6460-4112-9c8c-0165474b664f.mov

### Information

  - macOS Version: 12.4 (21F79)
  - Intel chip or Apple chip: Apple M1 Max
  - Docker Desktop Version: ~4.10.0 (82025)~ 4.10.1 (82475)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
❯ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
No fatal errors detected.
```

### Steps to reproduce the behavior

  1. Open Docker Desktop
  2. Go to Containers tab
  3. Disable a column
  4. Wait a couple of seconds
  5. The column will re-enable itself

**Comment by nicks** [comment]
thank you for reporting! we have a fix for this and it will be in the next release.

**Comment by nicks** [comment]
ya, likely 4.11

**Comment by gavin-timbrell** [comment]
We've just released [Docker Desktop 4.11.0](https://docs.docker.com/desktop/release-notes/#docker-desktop-4110) which includes a fix for this.
